zerver/models.py

```diff
@@ -2062,9 +2062,7 @@ def allowed_bot_types(self) -> List[int]:
         return allowed_bot_types
 
     def email_address_is_realm_public(self) -> bool:
-        if self.realm.email_address_visibility == Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
-            return True
-        if self.is_bot:
+        if self.email_address_visibility == UserProfile.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
```

Comment:
Maybe worth a small comment noting that bots always have `EMAIL_ADDRESS_VISIBILITY_EVERYONE`?

Reply:
Added it.